src/auspex/experiment.py

```diff
@@ -280,12 +280,24 @@ def update_descriptors(self):
             oc.update_descriptors()
 
     def declare_done(self):
+        while True:
+            all_done = True
+            for oc in self.output_connectors.values():
+                for os in oc.output_streams:
+                    # TODO: why does any queue interaction prevent adding out of order?
+                    if os.queue.qsize() > 0:
```

Comment:
Not implemented on macos

Reply:
I was having flakiness with this, but lets revert to empty. May have been due to other things